fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/RecordCursorTest.java

```diff
@@ -211,6 +211,29 @@ public void orElseTimingErrorTest() throws Exception {
         }
     }
 
+    @Test
+    public void asListWithContinuationTest() throws Exception {
+        final List<Integer> ints = IntStream.range(0, 50).boxed().collect(Collectors.toList());
+
+        final AtomicReference<RecordCursorResult<Integer>> finalResult = new AtomicReference<>();
+
+        int iterations = 0;
+        byte[] continuation = null;
+        do {
+            ++iterations;
+            List<Integer> values = RecordCursor.fromList(ints, continuation).limitRowsTo(10).asList(finalResult).get();
+            if (values.size() > 0) {
+                assertEquals(values.size(), 10);
+                assertEquals(values.get(0), (iterations - 1) * 10);
+                assertTrue(finalResult.get().getNoNextReason().isLimitReached());
+            }
```

Comment:
It would be good to assert about the final NoNextReason too
```suggestion
            } else {
                assertTrue(finalResult.get().getNoNextReason().isSourceExhausted());
            }
```